app/models/ems_refresh.rb

```diff
@@ -79,6 +92,66 @@ def self.refresh(target, id = nil)
     end
   end
 
+  # Rhev specific
+  def self.refresh_new_target(target_hash, ems_id)
+    ems = ExtManagementSystem.find(ems_id)
+
+    target = EmsRefresh.save_new_target(target_hash)
+    if target.nil?
+      _log.warn "Unknown target for event data: #{target_hash}."
+      return
+    end
+
+    ems.refresher.refresh(get_ar_objects(target))
+  end
+
+  def self.save_new_target(target_hash)
```

Comment:
Think `app/models/ems_refresh/save_inventory.rb` would be a better place for this